src/System.Management.Automation/DscSupport/JsonDscClassCache.cs

```diff
@@ -0,0 +1,3249 @@
+                    PsUtils.GetModuleManifestProperties(
+                        manifestPath,
+                        PsUtils.ManifestModuleVersionPropertyName);
+
+                object versionValue = dataFileSetting["ModuleVersion"];
+                if (versionValue is not null)
+                {
+                    Version moduleVersion;
+                    if (LanguagePrimitives.TryConvertTo(versionValue, out moduleVersion))
+                    {
+                        return new Tuple<string, Version>(moduleName, moduleVersion);
+                    }
+                    else
+                    {
+                        s_tracer.WriteLine(
+                            "DSC GetModuleVersion: ModuleVersion value '{0}' cannot be converted to System.Version. Skip the module '{1}'.",
+                            versionValue, moduleName);
+                    }
+                }
+                else
+                {
+                    s_tracer.WriteLine(
+                        "DSC GetModuleVersion: Manifest file '{0}' does not contain ModuleVersion. Skip the module '{1}'.",
+                        manifestPath, moduleName);
+                }
+            }
+            catch (PSInvalidOperationException ex)
+            {
+                s_tracer.WriteLine(
+                    "DSC GetModuleVersion: Error evaluating module manifest file '{0}', with error '{1}'. Skip the module '{2}'.",
+                    manifestPath, ex, moduleName);
+            }
+
+            return null;
+        }
+
+
+        private static void WriteWarning(string warning)
+        {
+            var executionContext = System.Management.Automation.Runspaces.Runspace.DefaultRunspace.ExecutionContext;
+            if (executionContext != null && executionContext.InternalHost != null && executionContext.InternalHost.UI != null)
+            {
+                executionContext.InternalHost.UI.WriteWarningLine(warning);
+            }
+        }
+
+        /// <summary>
+        /// Parses json file without adding it to caches or creating dynamic keywords.
+        /// </summary>
+        /// <param name="jsonFilePath">Path to json file</param>
+        /// <returns>List of classes from json file</returns>
+        public static IEnumerable<PSObject> ReadClassesFromJson(string jsonFilePath)
+        {
+            return ReadClassesFromJson(jsonFilePath, false);
+        }
+
+        /// <summary>
+        /// Parses json file without adding it to caches or creating dynamic keywords.
+        /// </summary>
+        /// <param name="jsonFilePath">Path to json file</param>
+        /// <param name="useNewRunspace">If True PowerShell will use a fresh runspace to do Json deserialization</param>
+        /// <returns>List of classes from json file</returns>
+        public static IEnumerable<PSObject> ReadClassesFromJson(string jsonFilePath, bool useNewRunspace)
+        {
+            if (string.IsNullOrEmpty(jsonFilePath))
```

Comment:
Also check if the jsonFilePath exists.

Reply:
If file doesn't exist then currently it returns `FileNotFoundException: Could not find file 'C:\MyClass.schema.json` - which seems to be the right thing to do.